### ISSUE TITLE
[UrT4.2] client authentication is bugged

### ISSUE THREAD

**Issue by danielepantaleone** [BigBrotherBot/big-brother-bot#343]
I'll report here a discussion I had with @thomasleveil related to a bug in the UrT4.2 parser authentication code. This is an extract of a b3.log showing the bug:

```
150627 21:43:45    DEBUG      'Client connected: ready to parse userinfo line'
150627 21:43:45    CONSOLE    '91:02 ClientUserinfo: 17 \\ip\\PLAYERIP:27960\\challenge\\-173954471\\qport\\37602\\protocol\\68\\name\\PLAYERNAME\\rate\\8000\\sex\\male\\handicap\\100\\color2\\5\\color1\\4\\snaps\\20\\authc\\10623\\authl\\playerauthlogin\\cl_guid\\D7B41871FC1A9C0989E93747FDA1FB39'
150627 21:43:45    VERBOS2    'Parsing userinfo: \\ip\\PLAYERIP:27960\\challenge\\-173954471\\qport\\37602\\protocol\\68\\name\\PLAYERNAME\\rate\\8000\\sex\\male\\handicap\\100\\color2\\5\\color1\\4\\snaps\\20\\authc\\10623\\authl\\playerauthlogin\\cl_guid\\D7B41871FC1A9C0989E93747FDA1FB39'
150627 21:43:45    VERBOSE    "Parsed user info: {'snaps': '20', 'protocol': '68', 'name': 'PLAYERNAME', 'cid': '17', 'handicap': '100', 'ip': 'PLAYERIP', 'challenge': '-173954471', 'sex': 'male', 'color1': '4', 'qport': '37602', 'rate': '8000', 'color2': '5', 'cl_guid': 'D7B41871FC1A9C0989E93747FDA1FB39', 'authl': 'playerauthlogin', 'authc': '10623', 'port': '27960'}"
150627 21:43:45    VERBOSE    "RCON sending (127.0.0.1:27923) 'auth-whois 17'"
150627 21:43:46    VERBOS2    "RCON: received 'auth: id: 0 - name: ^7PLAYERNAME - login: playerauthlogin - notoriety: basic - level: -1  \\n'"
150627 21:43:46    DEBUG      "Urt42 Client Connected: [17] PLAYERNAME - D7B41871FC1A9C0989E93747FDA1FB39 ({'name': 'PLAYERNAME', 'extra': None, 'level': '-1', 'notoriety': 'basic', 'cid': '0', 'login': 'playerauthlogin'})"
150627 21:43:46    VERBOS2    "Auth with {'ip': 'PLAYERIP', 'name': 'PLAYERNAME', 'pbid': 'playerauthlogin'}"
150627 21:43:46    VERBOSE    "Auth by FSA: 'playerauthlogin'"
150627 21:43:46    DEBUG      "Storage: getClientsMatching {'pbid': 'playerauthlogin'}"
150627 21:43:46    DEBUG      'Storage: getClient Client<@0:D7B41871FC1A9C0989E93747FDA1FB39|:"":None>'
150627 21:43:46    DEBUG      'Storage: getClient Client42<@13898:D7B41871FC1A9C0989E93747FDA1FB39|playerv:"PLAYERNAME":17>'
150627 21:43:46    DEBUG      'Storage: getClientAlias Alias(id=0, alias="PLAYERNAME", clientId=13898, numUsed=1)'
150627 21:43:46    DEBUG      'Storage: setClientAlias Alias(id=0, alias="PLAYERNAME", clientId=13898, numUsed=1)'
150627 21:43:46    DEBUG      "Storage: setClientAlias data {'num_used': 1, 'alias': 'PLAYERNAME', 'time_add': 1435434226, 'time_edit': 1435434226, 'client_id': 13898}"
150627 21:43:46    BOT        'New alias for 13898: PLAYERNAME'
150627 21:43:46    ERROR      u'Client has guid but its not the same D7B41871FC1A9C0989E93747FDA1FB39 <> 76CDE184F3BAB70F9729A1548A08B265'
150627 21:43:46    BOT        u"Client found in the storage @13898: welcome back PLAYERNAME` [FSA: 'playerauthlogin']"
```

As this log show, a player connect to the server, he's authorized using the FSA and then the parser checks also for guid match but the GUID being used by the client is apparently different from the one stored in the database (which can be caused by a player reinstalling the game onto another computer, without porting its qkey file but authenticating using FSA).

The bug is not juse that: by inspecting the database, the _spurious_ GUID seems not to be found anywhere:

```
mysql> select * from clients where guid='76CDE184F3BAB70F9729A1548A08B265';
Empty set (0.00 sec)
```

So basically this new GUID is never stored in the database together with the FSA, and B3 will keep logging this error. I suggest a rework of the authentication workflow for the iourt42 parser:

CURRENT: https://cacoo.com/diagrams/9x1Bcp3HrMaIwfr0
PROPOSAL: https://drive.google.com/file/d/0BwGkBOchEhbJNHRIMWhiLXhBTVk/view

**Comment by thomasleveil** [comment]
is this still troubling ?

**Comment by danielepantaleone** [comment]
Yes, I still have some log errors on my servers. I asked for logs from another server and they do not seem to have this problem, so maybe the error is due to inconsistency in my database table, but honestly I have no clue :disappointed:

**Comment by NOTMY** [comment]
Hello, I got those errors since at least year, used b3 dev versions from begining of this year with fresh mysql DB, used with dev from Jun 15 with fresh DB too and used latest dev from Oct 17 with DB from previous Jun 15. Now I see it only appears with clients using urt auth who got changed guid and b3 updating this old guid with new one so after when client connect again everything is fine but in previous version probably could be problem cus saw much more those errors but Im not sure if they were repeated and some clients basicly using scripts to delete/generate qkey/guid every time when run game again. Maybe in previous versions was problem with updating guids when auth was in use but now seems to work (got 2 servers and this player was on them both and only 1 error per server - maybe should change to 'warning' if works)  current version Oct 17 logs:

Gamelog old guid:
```
126:07 Session data initialised for client on slot 0 at 245232421
126:07 ClientConnect: 7
126:07 ClientUserinfo: 7 \ip\81.*.*.*:49471\challenge\331882357\qport\41541\protocol\68\name\PLAYER_NAME\racered\2\raceblue\2\rate\8000\ut_timenudge\0\cg_rgb\128 128 128\cg_physics\1\snaps\20\color1\4\color2\5\handicap\100\sex\male\cg_autoPickup\-1\cg_ghost\1\cl_time\2H+7ny1Va5f[b&=g\racefree\2\gear\ghAATAA\authc\29013\authl\PLAYER_AUTH\cl_guid\3829F99518ABA3A714FACACE6716FEC5\weapmodes\00000110220000020002000
126:07 ClientUserinfoChanged: 7 n\PLAYER_NAME\t\0\r\2\tl\0\f0\\f1\\f2\\a0\128\a1\128\a2\128
126:07 AccountValidated: 7 - PLAYER_AUTH - -1 - "basic"
126:07 ClientUserinfo: 7 \ip\81.*.*.*:49471\name\PLAYER_NAME\challenge\331882357\qport\41541\protocol\68\racered\2\raceblue\2\rate\8000\ut_timenudge\0\cg_rgb\128 128 128\cg_physics\1\snaps\20\color1\4\color2\5\handicap\100\sex\male\cg_autoPickup\-1\cg_ghost\1\cl_time\2H+7ny1Va5f[b&=g\racefree\2\gear\ghAATAA\authc\29013\authl\PLAYER_AUTH\cl_guid\3829F99518ABA3A714FACACE6716FEC5\weapmodes\00000110220000020002000
126:07 ClientUserinfoChanged: 7 n\PLAYER_NAME\t\0\r\2\tl\0\f0\\f1\\f2\\a0\128\a1\128\a2\128
```

Gamelog with changed guid:
```
163:31 Session data initialised for client on slot 0 at 237944687
163:31 ClientConnect: 0
163:31 ClientUserinfo: 0 \ip\81.*.*.*:56848\challenge\1545583743\qport\13596\protocol\68\name\PLAYER_NAME\racered\2\raceblue\2\rate\8000\ut_timenudge\0\cg_rgb\128 128 128\cg_physics\1\gear\ghAATAA\racefree\2\cl_time\FLbnf1?A&5(z|&5=\cg_ghost\1\cg_autoPickup\-1\sex\male\handicap\100\color2\5\color1\4\snaps\20\authc\18585\authl\PLAYER_AUTH\cl_guid\1BE887DFF5103CAAE22EC43F218DFFD7\weapmodes\00000110220000020002000
163:31 ClientUserinfoChanged: 0 n\PLAYER_NAME\t\0\r\2\tl\0\f0\\f1\\f2\\a0\128\a1\128\a2\128
163:31 AccountValidated: 0 - PLAYER_AUTH - -1 - "basic"
163:31 ClientUserinfo: 0 \ip\81.*.*.*:56848\name\PLAYER_NAME\challenge\1545583743\qport\13596\protocol\68\racered\2\raceblue\2\rate\8000\ut_timenudge\0\cg_rgb\128 128 128\cg_physics\1\gear\ghAATAA\racefree\2\cl_time\FLbnf1?A&5(z|&5=\cg_ghost\1\cg_autoPickup\-1\sex\male\handicap\100\color2\5\color1\4\snaps\20\authc\18585\authl\PLAYER_AUTH\cl_guid\1BE887DFF5103CAAE22EC43F218DFFD7\weapmodes\00000110220000020002000
163:31 ClientUserinfoChanged: 0 n\PLAYER_NAME\t\0\r\2\tl\0\f0\\f1\\f2\\a0\128\a1\128\a2\128
```

B3 log:
```
151202 23:30:32 CONSOLE  '163:31 Session data initialised for client on slot 0 at 237944687'
151202 23:30:32 VERBOSE  'Line did not match format: Session data initialised for client on slot 0 at 237944687'
151202 23:30:32 CONSOLE  '163:31 ClientConnect: 0'
151202 23:30:32 DEBUG    'Client connected: ready to parse userinfo line'
151202 23:30:32 CONSOLE  '163:31 ClientUserinfo: 0 \\ip\\81.*.*.*:56848\\challenge\\1545583743\\qport\\13596\\protocol\\68\\name\\PLAYER_NAME\\racered\\2\\raceblue\\2\\rate\\8000\\ut_timenudge\\0\\cg_rgb\\128 128 128\\cg_physics\\1\\gear\\ghAATAA\\racefree\\2\\cl_time\\FLbnf1?A&5(z|&5=\\cg_ghost\\1\\cg_autoPickup\\-1\\sex\\male\\handicap\\100\\color2\\5\\color1\\4\\snaps\\20\\authc\\18585\\authl\\PLAYER_AUTH\\cl_guid\\1BE887DFF5103CAAE22EC43F218DFFD7\\weapmodes\\00000110220000020002000'
151202 23:30:32 VERBOS2  'Parsing userinfo: \\ip\\81.*.*.*:56848\\challenge\\1545583743\\qport\\13596\\protocol\\68\\name\\PLAYER_NAME\\racered\\2\\raceblue\\2\\rate\\8000\\ut_timenudge\\0\\cg_rgb\\128 128 128\\cg_physics\\1\\gear\\ghAATAA\\racefree\\2\\cl_time\\FLbnf1?A&5(z|&5=\\cg_ghost\\1\\cg_autoPickup\\-1\\sex\\male\\handicap\\100\\color2\\5\\color1\\4\\snaps\\20\\authc\\18585\\authl\\PLAYER_AUTH\\cl_guid\\1BE887DFF5103CAAE22EC43F218DFFD7\\weapmodes\\00000110220000020002000'
151202 23:30:32 VERBOSE  "Parsed user info: {'racered': '2', 'protocol': '68', 'cg_ghost': '1', 'ip': '81.*.*.*', 'sex': 'male', 'racefree': '2', 'rate': '8000', 'port': '56848', 'cl_time': 'FLbnf1?A&5(z|&5=', 'weapmodes': '00000110220000020002000', 'cl_guid': '1BE887DFF5103CAAE22EC43F218DFFD7', 'authl': 'PLAYER_AUTH', 'authc': '18585', 'snaps': '20', 'cg_rgb': '128 128 128', 'gear': 'ghAATAA', 'handicap': '100', 'cg_autoPickup': '-1', 'color1': '4', 'qport': '13596', 'color2': '5', 'cg_physics': '1', 'name': 'PLAYER_NAME', 'cid': '0', 'challenge': '1545583743', 'ut_timenudge': '0', 'raceblue': '2'}"
151202 23:30:32 DEBUG    'Urt42 Client Connected: [0] PLAYER_NAME - 1BE887DFF5103CAAE22EC43F218DFFD7 ({})'
151202 23:30:32 VERBOSE  "Auth with {'ip': '81.*.*.*', 'guid': '1BE887DFF5103CAAE22EC43F218DFFD7', 'name': 'PLAYER_NAME', 'pbid': 'PLAYER_AUTH'}"
151202 23:30:32 DEBUG    "Auth by FSA: 'PLAYER_AUTH'"
151202 23:30:32 DEBUG    "Storage: getClientsMatching {'pbid': 'PLAYER_AUTH'}"
151202 23:30:32 DEBUG    'Storage: getClient Client42<@0:1BE887DFF5103CAAE22EC43F218DFFD7|:"":None>'
151202 23:30:32 DEBUG    'Storage: getClient Client42<@480:1BE887DFF5103CAAE22EC43F218DFFD7|PLAYER_AUTH:"PLAYER_NAME":0>'
151202 23:30:32 VERBOS2  'Aborted making alias for cid 0: name is the same'
151202 23:30:32 ERROR    u'Client has guid but its not the same 1BE887DFF5103CAAE22EC43F218DFFD7 <> 3829F99518ABA3A714FACACE6716FEC5'
151202 23:30:32 BOT      u"Client found in the storage @480: welcome back PLAYER_NAME [FSA: 'PLAYER_AUTH']"
151202 23:30:32 VERBOS2  'Aborted making alias for cid 0: name is the same'
151202 23:30:32 DEBUG    'Storage: setClient Client42<@480:1BE887DFF5103CAAE22EC43F218DFFD7|PLAYER_AUTH:"PLAYER_NAME":0>'
151202 23:30:32 DEBUG    "Storage: setClient data {'time_add': 1445708517, 'password': u'', 'time_edit': 1449095432, 'name': 'PLAYER_NAME', 'group_bits': 0, 'ip': '81.*.*.*', 'auto_login': 1, 'greeting': u'', 'pbid': 'PLAYER_AUTH', 'connections': 9, 'mask_level': 0, 'login': u'', 'guid': '1BE887DFF5103CAAE22EC43F218DFFD7', 'id': 480}"
151202 23:30:32 DEBUG    "Client authorized: PLAYER_NAME [@480] [GUID: '1BE887DFF5103CAAE22EC43F218DFFD7'] [FSA: 'PLAYER_AUTH']"
151202 23:30:32 VERBOSE  'Queueing event Client Authenticated : Client42<@480:1BE887DFF5103CAAE22EC43F218DFFD7|PLAYER_AUTH:"PLAYER_NAME":0>'
151202 23:30:32 VERBOSE  'Parsing event: Client Authenticated: IpbanPlugin'
151202 23:30:32 DEBUG    'IpbanPlugin: checking player: <cid:0,name:PLAYER_NAME,ip:81.*.*.*>'
151202 23:30:32 DEBUG    'IpbanPlugin: client accepted (no active Ban/TempBan found): <cid:0,name:PLAYER_NAME,ip:81.*.*.*>'
151202 23:30:32 VERBOS2  'Client Authenticated event handled by IpbanPlugin in 1.362 ms'
151202 23:30:32 VERBOSE  'Parsing event: Client Authenticated: HelloPlugin'
151202 23:30:32 VERBOS2  'Client Authenticated event handled by HelloPlugin in 6.693 ms'
151202 23:30:32 CONSOLE  '163:31 ClientUserinfoChanged: 0 n\\PLAYER_NAME\\t\\0\\r\\2\\tl\\0\\f0\\\\f1\\\\f2\\\\a0\\128\\a1\\128\\a2\\128'
151202 23:30:32 VERBOS2  'Parsing userinfo: \\n\\PLAYER_NAME\\t\\0\\r\\2\\tl\\0\\f0\\\\f1\\\\f2\\\\a0\\128\\a1\\128\\a2\\128'
151202 23:30:32 VERBOSE  "Parsed userinfo: {'cid': '0', 'n': 'PLAYER_NAME', 'a1': '128', 'a0': '128', 'tl': '0', 'r': '2', 't': '0', 'a2': '128'}"
151202 23:30:32 VERBOS2  'Aborted making alias for cid 0: name is the same'
151202 23:30:32 CONSOLE  '163:31 AccountValidated: 0 - PLAYER_AUTH - -1 - "basic"'
151202 23:30:32 CONSOLE  '163:31 ClientUserinfo: 0 \\ip\\81.*.*.*:56848\\name\\PLAYER_NAME\\challenge\\1545583743\\qport\\13596\\protocol\\68\\racered\\2\\raceblue\\2\\rate\\8000\\ut_timenudge\\0\\cg_rgb\\128 128 128\\cg_physics\\1\\gear\\ghAATAA\\racefree\\2\\cl_time\\FLbnf1?A&5(z|&5=\\cg_ghost\\1\\cg_autoPickup\\-1\\sex\\male\\handicap\\100\\color2\\5\\color1\\4\\snaps\\20\\authc\\18585\\authl\\PLAYER_AUTH\\cl_guid\\1BE887DFF5103CAAE22EC43F218DFFD7\\weapmodes\\00000110220000020002000'
151202 23:30:32 VERBOS2  'Parsing userinfo: \\ip\\81.*.*.*:56848\\name\\PLAYER_NAME\\challenge\\1545583743\\qport\\13596\\protocol\\68\\racered\\2\\raceblue\\2\\rate\\8000\\ut_timenudge\\0\\cg_rgb\\128 128 128\\cg_physics\\1\\gear\\ghAATAA\\racefree\\2\\cl_time\\FLbnf1?A&5(z|&5=\\cg_ghost\\1\\cg_autoPickup\\-1\\sex\\male\\handicap\\100\\color2\\5\\color1\\4\\snaps\\20\\authc\\18585\\authl\\PLAYER_AUTH\\cl_guid\\1BE887DFF5103CAAE22EC43F218DFFD7\\weapmodes\\00000110220000020002000'
151202 23:30:32 VERBOSE  "Parsed user info: {'racered': '2', 'protocol': '68', 'cg_ghost': '1', 'ip': '81.*.*.*', 'sex': 'male', 'racefree': '2', 'rate': '8000', 'port': '56848', 'cl_time': 'FLbnf1?A&5(z|&5=', 'weapmodes': '00000110220000020002000', 'cl_guid': '1BE887DFF5103CAAE22EC43F218DFFD7', 'authl': 'PLAYER_AUTH', 'authc': '18585', 'snaps': '20', 'cg_rgb': '128 128 128', 'gear': 'ghAATAA', 'handicap': '100', 'cg_autoPickup': '-1', 'color1': '4', 'qport': '13596', 'color2': '5', 'cg_physics': '1', 'name': 'PLAYER_NAME', 'cid': '0', 'challenge': '1545583743', 'ut_timenudge': '0', 'raceblue': '2'}"
151202 23:30:32 VERBOS2  'Aborted making alias for cid 0: name is the same'
151202 23:30:33 CONSOLE  '163:31 ClientUserinfoChanged: 0 n\\PLAYER_NAME\\t\\0\\r\\2\\tl\\0\\f0\\\\f1\\\\f2\\\\a0\\128\\a1\\128\\a2\\128'
151202 23:30:33 VERBOS2  'Parsing userinfo: \\n\\PLAYER_NAME\\t\\0\\r\\2\\tl\\0\\f0\\\\f1\\\\f2\\\\a0\\128\\a1\\128\\a2\\128'
151202 23:30:33 VERBOSE  "Parsed userinfo: {'cid': '0', 'n': 'PLAYER_NAME', 'a1': '128', 'a0': '128', 'tl': '0', 'r': '2', 't': '0', 'a2': '128'}"
151202 23:30:33 VERBOS2  'Aborted making alias for cid 0: name is the same'
151202 23:30:33 VERBOSE  'RCON sending (127.0.0.1:2015) "^3PLAYER_NAME^7^7\'s last visit ^28 ^7was ^36 days, 1 hour, 44 minutes ^7ago"'
151202 23:30:33 VERBOS2  'RCON: received \'broadcast: print "^7server:^3 ^3PLAYER_NAME^7^7\\\'s last visit ^28 ^7was ^36 days, 1 hour, 44 minutes ^7ago"\\n\''
151202 23:30:33 VERBOSE  "RCON sending (127.0.0.1:2015) '^7#480 ^3PLAYER_NAME^7^7 from ^5Belgium'"
151202 23:30:33 VERBOS2  'RCON: received \'broadcast: print "^7server:^3 ^7#480 ^3PLAYER_NAME^7^7 from ^5Belgium"\\n\''
151202 23:30:39 CONSOLE  '163:38 ClientDisconnect: 0'
151202 23:30:39 VERBOSE  'Queueing event Client Disconnect : 0'
151202 23:30:39 VERBOSE  'Parsing event: Client Disconnect: HelloPlugin'
151202 23:30:39 VERBOS2  'Client Disconnect event handled by HelloPlugin in 0.740 ms'
151202 23:30:45 CONSOLE  '163:44 Session data initialised for client on slot 0 at 237957010'
151202 23:30:45 VERBOSE  'Line did not match format: Session data initialised for client on slot 0 at 237957010'
151202 23:30:45 CONSOLE  '163:44 ClientConnect: 0'
151202 23:30:45 DEBUG    'Client connected: ready to parse userinfo line'
151202 23:30:45 CONSOLE  '163:44 ClientUserinfo: 0 \\ip\\81.*.*.*:56848\\challenge\\1545583743\\qport\\13596\\protocol\\68\\name\\PLAYER_NAME\\racered\\2\\raceblue\\2\\rate\\8000\\ut_timenudge\\0\\cg_rgb\\128 128 128\\cg_physics\\1\\gear\\ghAATAA\\racefree\\2\\cl_time\\FLbnf1?A&5(z|&5=\\cg_ghost\\1\\cg_autoPickup\\-1\\sex\\male\\handicap\\100\\color2\\5\\color1\\4\\snaps\\20\\authc\\18585\\authl\\PLAYER_AUTH\\cl_guid\\1BE887DFF5103CAAE22EC43F218DFFD7\\weapmodes\\00000110220000020002000'
151202 23:30:45 VERBOS2  'Parsing userinfo: \\ip\\81.*.*.*:56848\\challenge\\1545583743\\qport\\13596\\protocol\\68\\name\\PLAYER_NAME\\racered\\2\\raceblue\\2\\rate\\8000\\ut_timenudge\\0\\cg_rgb\\128 128 128\\cg_physics\\1\\gear\\ghAATAA\\racefree\\2\\cl_time\\FLbnf1?A&5(z|&5=\\cg_ghost\\1\\cg_autoPickup\\-1\\sex\\male\\handicap\\100\\color2\\5\\color1\\4\\snaps\\20\\authc\\18585\\authl\\PLAYER_AUTH\\cl_guid\\1BE887DFF5103CAAE22EC43F218DFFD7\\weapmodes\\00000110220000020002000'
151202 23:30:45 VERBOSE  "Parsed user info: {'racered': '2', 'protocol': '68', 'cg_ghost': '1', 'ip': '81.*.*.*', 'sex': 'male', 'racefree': '2', 'rate': '8000', 'port': '56848', 'cl_time': 'FLbnf1?A&5(z|&5=', 'weapmodes': '00000110220000020002000', 'cl_guid': '1BE887DFF5103CAAE22EC43F218DFFD7', 'authl': 'PLAYER_AUTH', 'authc': '18585', 'snaps': '20', 'cg_rgb': '128 128 128', 'gear': 'ghAATAA', 'handicap': '100', 'cg_autoPickup': '-1', 'color1': '4', 'qport': '13596', 'color2': '5', 'cg_physics': '1', 'name': 'PLAYER_NAME', 'cid': '0', 'challenge': '1545583743', 'ut_timenudge': '0', 'raceblue': '2'}"
151202 23:30:45 DEBUG    'Urt42 Client Connected: [0] PLAYER_NAME - 1BE887DFF5103CAAE22EC43F218DFFD7 ({})'
151202 23:30:45 VERBOSE  "Auth with {'ip': '81.*.*.*', 'guid': '1BE887DFF5103CAAE22EC43F218DFFD7', 'name': 'PLAYER_NAME', 'pbid': 'PLAYER_AUTH'}"
151202 23:30:45 DEBUG    "Auth by FSA: 'PLAYER_AUTH'"
151202 23:30:45 DEBUG    "Storage: getClientsMatching {'pbid': 'PLAYER_AUTH'}"
151202 23:30:45 DEBUG    'Storage: getClient Client42<@0:1BE887DFF5103CAAE22EC43F218DFFD7|:"":None>'
151202 23:30:45 DEBUG    'Storage: getClient Client42<@480:1BE887DFF5103CAAE22EC43F218DFFD7|PLAYER_AUTH:"PLAYER_NAME":0>'
151202 23:30:45 VERBOS2  'Aborted making alias for cid 0: name is the same'
151202 23:30:45 BOT      u"Client found in the storage @480: welcome back PLAYER_NAME [FSA: 'PLAYER_AUTH']"
151202 23:30:45 VERBOS2  'Aborted making alias for cid 0: name is the same'
151202 23:30:45 DEBUG    'Storage: setClient Client42<@480:1BE887DFF5103CAAE22EC43F218DFFD7|PLAYER_AUTH:"PLAYER_NAME":0>'
151202 23:30:45 DEBUG    "Storage: setClient data {'time_add': 1445708517, 'password': u'', 'time_edit': 1449095445, 'name': 'PLAYER_NAME', 'group_bits': 0, 'ip': '81.*.*.*', 'auto_login': 1, 'greeting': u'', 'pbid': 'PLAYER_AUTH', 'connections': 10, 'mask_level': 0, 'login': u'', 'guid': '1BE887DFF5103CAAE22EC43F218DFFD7', 'id': 480}"
151202 23:30:45 DEBUG    "Client authorized: PLAYER_NAME [@480] [GUID: '1BE887DFF5103CAAE22EC43F218DFFD7'] [FSA: 'PLAYER_AUTH']"
151202 23:30:45 VERBOSE  'Queueing event Client Authenticated : Client42<@480:1BE887DFF5103CAAE22EC43F218DFFD7|PLAYER_AUTH:"PLAYER_NAME":0>'
151202 23:30:45 VERBOSE  'Parsing event: Client Authenticated: IpbanPlugin'
151202 23:30:45 DEBUG    'IpbanPlugin: checking player: <cid:0,name:PLAYER_NAME,ip:81.*.*.*>'
151202 23:30:45 DEBUG    'IpbanPlugin: client accepted (no active Ban/TempBan found): <cid:0,name:PLAYER_NAME,ip:81.*.*.*>'
151202 23:30:45 VERBOS2  'Client Authenticated event handled by IpbanPlugin in 1.388 ms'
151202 23:30:45 VERBOSE  'Parsing event: Client Authenticated: HelloPlugin'
151202 23:30:45 VERBOS2  'Client Authenticated event handled by HelloPlugin in 1.350 ms'
151202 23:30:45 CONSOLE  '163:44 ClientUserinfoChanged: 0 n\\PLAYER_NAME\\t\\0\\r\\2\\tl\\0\\f0\\\\f1\\\\f2\\\\a0\\128\\a1\\128\\a2\\128'
151202 23:30:45 VERBOS2  'Parsing userinfo: \\n\\PLAYER_NAME\\t\\0\\r\\2\\tl\\0\\f0\\\\f1\\\\f2\\\\a0\\128\\a1\\128\\a2\\128'
151202 23:30:45 VERBOSE  "Parsed userinfo: {'cid': '0', 'n': 'PLAYER_NAME', 'a1': '128', 'a0': '128', 'tl': '0', 'r': '2', 't': '0', 'a2': '128'}"
151202 23:30:45 VERBOS2  'Aborted making alias for cid 0: name is the same'
151202 23:30:45 VERBOSE  'RCON sending (127.0.0.1:2015) "^3PLAYER_NAME^7^7\'s last visit ^29 ^7was ^313 seconds ^7ago"'
151202 23:30:45 CONSOLE  '163:44 AccountValidated: 0 - PLAYER_AUTH - -1 - "basic"'
151202 23:30:45 CONSOLE  '163:44 ClientUserinfo: 0 \\ip\\81.*.*.*:56848\\name\\PLAYER_NAME\\challenge\\1545583743\\qport\\13596\\protocol\\68\\racered\\2\\raceblue\\2\\rate\\8000\\ut_timenudge\\0\\cg_rgb\\128 128 128\\cg_physics\\1\\gear\\ghAATAA\\racefree\\2\\cl_time\\FLbnf1?A&5(z|&5=\\cg_ghost\\1\\cg_autoPickup\\-1\\sex\\male\\handicap\\100\\color2\\5\\color1\\4\\snaps\\20\\authc\\18585\\authl\\PLAYER_AUTH\\cl_guid\\1BE887DFF5103CAAE22EC43F218DFFD7\\weapmodes\\00000110220000020002000'
151202 23:30:45 VERBOS2  'Parsing userinfo: \\ip\\81.*.*.*:56848\\name\\PLAYER_NAME\\challenge\\1545583743\\qport\\13596\\protocol\\68\\racered\\2\\raceblue\\2\\rate\\8000\\ut_timenudge\\0\\cg_rgb\\128 128 128\\cg_physics\\1\\gear\\ghAATAA\\racefree\\2\\cl_time\\FLbnf1?A&5(z|&5=\\cg_ghost\\1\\cg_autoPickup\\-1\\sex\\male\\handicap\\100\\color2\\5\\color1\\4\\snaps\\20\\authc\\18585\\authl\\PLAYER_AUTH\\cl_guid\\1BE887DFF5103CAAE22EC43F218DFFD7\\weapmodes\\00000110220000020002000'
151202 23:30:45 VERBOSE  "Parsed user info: {'racered': '2', 'protocol': '68', 'cg_ghost': '1', 'ip': '81.*.*.*', 'sex': 'male', 'racefree': '2', 'rate': '8000', 'port': '56848', 'cl_time': 'FLbnf1?A&5(z|&5=', 'weapmodes': '00000110220000020002000', 'cl_guid': '1BE887DFF5103CAAE22EC43F218DFFD7', 'authl': 'PLAYER_AUTH', 'authc': '18585', 'snaps': '20', 'cg_rgb': '128 128 128', 'gear': 'ghAATAA', 'handicap': '100', 'cg_autoPickup': '-1', 'color1': '4', 'qport': '13596', 'color2': '5', 'cg_physics': '1', 'name': 'PLAYER_NAME', 'cid': '0', 'challenge': '1545583743', 'ut_timenudge': '0', 'raceblue': '2'}"
151202 23:30:45 VERBOS2  'Aborted making alias for cid 0: name is the same'
151202 23:30:45 CONSOLE  '163:44 ClientUserinfoChanged: 0 n\\PLAYER_NAME\\t\\0\\r\\2\\tl\\0\\f0\\\\f1\\\\f2\\\\a0\\128\\a1\\128\\a2\\128'
151202 23:30:45 VERBOS2  'Parsing userinfo: \\n\\PLAYER_NAME\\t\\0\\r\\2\\tl\\0\\f0\\\\f1\\\\f2\\\\a0\\128\\a1\\128\\a2\\128'
151202 23:30:45 VERBOSE  "Parsed userinfo: {'cid': '0', 'n': 'PLAYER_NAME', 'a1': '128', 'a0': '128', 'tl': '0', 'r': '2', 't': '0', 'a2': '128'}"
151202 23:30:45 VERBOS2  'Aborted making alias for cid 0: name is the same'
151202 23:30:45 VERBOS2  'RCON: received \'broadcast: print "^7server:^3 ^3PLAYER_NAME^7^7\\\'s last visit ^29 ^7was ^313 seconds ^7ago"\\n\''
151202 23:30:45 VERBOSE  "RCON sending (127.0.0.1:2015) '^7#480 ^3PLAYER_NAME^7^7 from ^5Belgium'"
151202 23:30:45 VERBOS2  'RCON: received \'broadcast: print "^7server:^3 ^7#480 ^3PLAYER_NAME^7^7 from ^5Belgium"\\n\''
```

But ye second time client connect again right after map download.
Sorry for maybe too much logs.
